extensions/opslevel/src/clone-action.tsx

```diff
@@ -0,0 +1,118 @@
+}
+
+function repoInfo(checkoutLocation: string, repoUrl: string) {
+    const repoName = repoUrl.split("/").pop();
+    const repoPath = `${checkoutLocation}/${repoName}`;
+    return { repoName, repoPath };
+}
+
+async function folderExists(folderPath: string) {
+    try {
+        await fs.access(folderPath);
+        return true;
+    } catch (err) {
+        console.error(err);
+        return false;
+    }
+}
+
+async function selectEditor(editorBundleId: string, repoPath: string, repoUrl: string) {
+    await setEditor(repoPath, editorBundleId);
+
+    if (!(await folderExists(repoPath))) {
+        Clipboard.copy(`git clone ${repoUrl} ${repoPath}`);
+        showHUD("Repository not cloned. Clone command copied to clipboard");
+    }
+
+    try {
+        await open(repoPath, editorBundleId);
+    } catch (error) {
+        showToast({
+            style: Toast.Style.Failure,
+            title: "Failed to open editor",
+            message: error instanceof Error ? error.message : String(error),
+        });
+    }
+}
+
+export default function OpenRepoInEditorAction({
+    repoUrl,
+    onVisisted,
+}: {
+    repoUrl: string | undefined | null;
+    onVisisted: () => void;
+}) {
+    if (!repoUrl) return null;
+
+    const { repoPath } = repoInfo(checkoutLocation, repoUrl);
+    const { isLoading: isEditorLoading, data: editorBundleId } = usePromise(
+        async (rp: string) => await getEditor(rp),
+        [repoPath]
+    );
+    const { isLoading: isApplicationsLoading, data: applications } = usePromise(
+        async () => await getApplications(),
+        []
+    );
+
+    return (
+        <ActionPanel.Section>
+            {isEditorLoading || isApplicationsLoading ? (
+                <Action title="Loading..." />
+            ) : (
+                [
+                    editorBundleId ? (
+                        <Action
+                            key="open-in-editor"
+                            // eslint-disable-next-line @raycast/prefer-title-case
+                            title={`Open In ${
+                                applications?.find((app) => app.bundleId === editorBundleId)?.name ?? "Unknown"
+                            }`}
+                            shortcut={{ modifiers: ["cmd"], key: "o" }}
+                            onAction={async () => {
+                                onVisisted();
+                                await selectEditor(editorBundleId, repoPath, repoUrl);
+                            }}
+                        />
+                    ) : null,
+                    <ActionPanel.Submenu
+                        key="open-in-choice"
+                        title="Open in ..."
+                        isLoading={isApplicationsLoading}
+                        onOpen={onVisisted}
+                    >
+                        {applications?.map((app) => (
+                            <Action
+                                key={`open-in-${app.bundleId}`}
```

Comment:
You can use [FileIcon](https://developers.raycast.com/api-reference/user-interface/icons-and-images#fileicon) (`icon={{ fileIcon: app.path }}`) to show the app's icon :D

Reply:
Great trick!